src/js/builtins/Module.ts

```diff
@@ -34,45 +38,70 @@ export function overridableRequire(this: CommonJSModuleRecord, id: string) {
     $evaluateCommonJSModule(existing);
     return existing.exports;
   }
-
-  if (id.endsWith(".node")) {
+  const filename = id;
+  if (filename.endsWith(".node")) {
     return $internalRequire(id);
   }
-
   // To handle import/export cycles, we need to create a module object and put
   // it into the map before we import it.
   const mod = $createCommonJSModule(id, {}, false, this);
   $requireMap.$set(id, mod);
 
+  // @ts-ignore - TS doesn't know each function is isolated.
+  const Module = require("node:module");
```

Comment:
this call is going to make every module load slower. we need to find a better place to do this.

Reply:
My understanding is that `const Module = require("node:module")` is hoisted (or cached in some way through magic transformation) 🤞

Reply:
Perf concern covered https://github.com/oven-sh/bun/pull/8193#discussion_r1452960872.